src/vs/workbench/browser/parts/editor/editorActions.ts

```diff
@@ -2696,3 +2697,32 @@ export class NewEmptyEditorWindowAction extends Action2 {
 		auxiliaryEditorPart.activeGroup.focus();
 	}
 }
+
+export class ToggleOvertypeInsertMode extends Action2 {
+
+	constructor() {
+		super({
+			id: 'editor.action.toggleOvertypeInsertMode',
+			title: {
+				...localize2('toggleOvertypeInsertMode', "Toggle Overtype/Insert Mode"),
+				mnemonicTitle: localize({ key: 'mitoggleOvertypeInsertMode', comment: ['&& denotes a mnemonic'] }, "&&Toggle Overtype/Insert Mode"),
+			},
+			metadata: {
+				description: localize2('toggleOvertypeMode.description', "Toggle between overtype and insert mode"),
+			},
+			keybinding: {
+				weight: KeybindingWeight.WorkbenchContrib,
+				primary: KeyCode.Insert,
+				secondary: [KeyMod.Shift | KeyCode.Ctrl | KeyCode.KeyQ],
```

Comment:
Sublime text uses `alt + cmd + o` to toggle overtype mode, which we can mimic. I will change the keybinding to this.

Vim and MacVim use `shift+r` but this would not work for us

Reply:
What about the keybinding that works in most GUI editors? [Insert] ?

Reply:
`alt + cmd + o` will be used on mac only, since mac does not have the insert key. Windows based systems will have the insert keybinding.